apps/dotcom/sync-worker/src/TLUserDurableObject.ts

```diff
@@ -283,80 +284,86 @@ export class TLUserDurableObject extends DurableObject<Environment> {
-										}
-									} else if (
-										currentFile &&
-										currentFile.published &&
-										currentFile.lastPublished < rest.lastPublished
-									) {
+					}
+					case 'update': {
+						const mutableColumns = Object.keys(update.row).filter((k) =>
+							isColumnMutable(update.table, k)
+						)
+						if (mutableColumns.length === 0) continue
+						const updates = Object.fromEntries(
+							mutableColumns.map((k) => [k, (update.row as any)[k]])
+						)
+						if (update.table === 'file_state') {
+							const { fileId, userId } = update.row as any
+							await sql`update public.file_state set ${sql(updates)} where "fileId" = ${fileId} and "userId" = ${userId}`
+						} else {
+							const { id, ...rest } = update.row as any
+
+							await sql`update ${sql('public.' + update.table)} set ${sql(updates)} where id = ${id}`
+							if (update.table === 'file') {
+								const currentFile = this.cache.store.getFullData()?.files.find((f) => f.id === id)
+								if (currentFile && currentFile.published !== rest.published) {
+									if (rest.published) {
 										await this.publishSnapshot(currentFile)
+									} else {
+										await this.unpublishSnapshot(currentFile)
 									}
+								} else if (
+									currentFile &&
+									currentFile.published &&
+									currentFile.lastPublished < rest.lastPublished
+								) {
+									await this.publishSnapshot(currentFile)
 								}
 							}
-							break
 						}
-						case 'delete':
-							if (update.table === 'file_state') {
-								const { fileId, userId } = update.row as any
-								await sql`delete from public.file_state where "fileId" = ${fileId} and "userId" = ${userId}`
-							} else {
-								const { id } = update.row as any
-								await sql`delete from ${sql('public.' + update.table)} where id = ${id}`
-							}
-							if (update.table === 'file') {
-								const { id } = update.row as TlaFile
-								await this.deleteFileStuff(id)
-							}
-							break
+						break
 					}
-					this.cache.store.updateOptimisticData([update], msg.mutationId)
+					case 'delete':
+						if (update.table === 'file_state') {
+							const { fileId, userId } = update.row as any
+							await sql`delete from public.file_state where "fileId" = ${fileId} and "userId" = ${userId}`
+						} else {
+							const { id } = update.row as any
+							await sql`delete from ${sql('public.' + update.table)} where id = ${id}`
+						}
+						if (update.table === 'file') {
+							const { id } = update.row as TlaFile
+							await this.deleteFileStuff(id)
+						}
+						break
 				}
-			})
+				this.cache.store.updateOptimisticData([update], msg.mutationId)
+			}
+		})
+	}
 
+	private async handleMutate(msg: ZClientSentMessage) {
+		this.assertCache()
+		try {
+			// we connect to pg via a pooler, so in the case that the pool is exhausted
+			// we need to retry the connection. (also in the case that a neon branch is asleep apparently?)
```

Comment:
I reached out and got a fast first reply from them asking for more detailed information about when the issues occurred. I've send them the exact times of all the errors I found in sentry 🤞